homeassistant/components/conversation/default_agent.py

```diff
@@ -81,16 +83,24 @@ def async_setup(hass: core.HomeAssistant) -> None:
         async_should_expose(hass, DOMAIN, entity_id)
 
     @core.callback
-    def async_handle_entity_registry_changed(event: core.Event) -> None:
-        """Set expose flag on newly created entities."""
-        if event.data["action"] == "create":
-            async_should_expose(hass, DOMAIN, event.data["entity_id"])
+    def async_entity_state_listener(
+        changed_entity: str,
+        old_state: core.State | None,
+        new_state: core.State | None,
+    ):
+        """Set expose flag on new entities."""
+        if old_state is not None or new_state is None:
+            return
+        async_should_expose(hass, DOMAIN, changed_entity)
+
+    @core.callback
+    def async_hass_started(hass: core.HomeAssistant) -> None:
+        """Set expose flag on all entities."""
+        for state in hass.states.async_all():
+            async_should_expose(hass, DOMAIN, state.entity_id)
+        async_track_state_change(hass, MATCH_ALL, async_entity_state_listener)
```

Comment:
For now that's ok, but in a future PR we should probably listen directly using `hass.events.async_listen` so we could pass a `filter` function